### PR TITLE
Add CFBenchmark rest transport v3 EA

### PR DESCRIPTION
**Pull request by bogos**
[PDI-1954](https://smartcontract-it.atlassian.net/browse/PDI-1954)

[PDI-1954]: https://smartcontract-it.atlassian.net/browse/PDI-1954?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/sources/cfbenchmarks-test/src/endpoint/common/crypto.ts [hunk 1]
```diff
@@ -76,21 +76,19 @@ export const cryptoRequestTransform = (req: AdapterRequest<RequestParams>): void
...
-    // }
+    if (config?.API_SECONDARY) {
+      if (config?.WS_ENABLED) return 'websocketSecondary'
+      return 'restSecondary'
+    }
 
-    // if (config?.WS_ENABLED) return 'websocket'
-    // return 'rest'
-    return config?.API_SECONDARY ? 'websocketSecondary' : 'websocket'
+    if (config?.WS_ENABLED) return 'websocket'
```
Comment (boxhock) [inline packages/sources/cfbenchmarks-test/src/endpoint/common/crypto.ts:90]: Remove `?` in the latest version of the EA framework

File: packages/sources/cfbenchmarks-test/test/integration/__snapshots__/adapter.test.ts.snap [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+exports[`rest crypto endpoint should return success 1`] = `
+Object {
+  "data": Object {
+    "result": 39829.3,
+  },
+  "result": 39829.3,
+  "statusCode": 200,
+  "timestamps": Object {
+    "providerDataReceived": Any<Number>,
+    "providerDataRequested": Any<Number>,
```
Comment (boxhock) [inline packages/sources/cfbenchmarks-test/test/integration/__snapshots__/adapter.test.ts.snap:12]: Doesn't seem right

File: packages/sources/cfbenchmarks-test/src/config/index.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ export const customSettings = {
     description: 'Toggle to set the default endpoint to use WebSockets',
     type: 'boolean',
     required: false,
-    default: true,
+    default: false,
```
Comment (boxhock) [inline packages/sources/cfbenchmarks-test/src/config/index.ts:47]: Can you check the validation method below? Seems like that would prevent this?

File: .changeset/quick-chicken-enjoy.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@chainlink/cfbenchmarks-test-adapter': patch
```
Comment (boxhock) [inline .changeset/quick-chicken-enjoy.md:2]:
```suggestion
'@chainlink/cfbenchmarks-test-adapter': minor
```

**Review by boxhock — CHANGES_REQUESTED** [review]

File: .changeset/quick-chicken-enjoy.md (hunk 4 above)
Comment (bogos) [reply .changeset/quick-chicken-enjoy.md]: Done

File: packages/sources/cfbenchmarks-test/src/config/index.ts (hunk 3 above)
Comment (bogos) [reply packages/sources/cfbenchmarks-test/src/config/index.ts]: Done

File: packages/sources/cfbenchmarks-test/src/endpoint/common/crypto.ts (hunk 1 above)
Comment (bogos) [reply packages/sources/cfbenchmarks-test/src/endpoint/common/crypto.ts]: Done

File: packages/sources/cfbenchmarks-test/test/integration/__snapshots__/adapter.test.ts.snap (hunk 2 above)
Comment (bogos) [reply packages/sources/cfbenchmarks-test/test/integration/__snapshots__/adapter.test.ts.snap]: Done

File: packages/sources/cfbenchmarks-test/src/config/index.ts [hunk 5]
```diff
@@ -44,10 +44,9 @@ export const customSettings = {
     description: 'Toggle to set the default endpoint to use WebSockets',
     type: 'boolean',
     required: false,
-    default: true,
     validate: (value?: boolean) => {
       if (!value) {
-        return 'WS_ENABLED must be set to true. Non-WS endpoints are not supported yet.'
+        return 'WS_ENABLED must be set to true'
```
Comment (boxhock) [inline packages/sources/cfbenchmarks-test/src/config/index.ts:49]: > WS_ENABLED must be set to true

Why?

File: packages/sources/cfbenchmarks-test/src/endpoint/rest/crypto.ts [hunk 6]
```diff
@@ -21,15 +21,23 @@ export type RestEndpointTypes = EndpointTypes & {
 // NOTE: This is using the BatchWarming transport, but the actual API endpoint is not batchable
```
Comment (boxhock) [inline packages/sources/cfbenchmarks-test/src/endpoint/rest/crypto.ts:21]: This comment should probably be updated

**Review by boxhock — CHANGES_REQUESTED** [review]

File: packages/sources/cfbenchmarks-test/src/endpoint/rest/crypto.ts (hunk 6 above)
Comment (bogos) [reply packages/sources/cfbenchmarks-test/src/endpoint/rest/crypto.ts:21]: Comment removed

File: packages/sources/cfbenchmarks-test/src/config/index.ts (hunk 5 above)
Comment (amit-momin) [reply packages/sources/cfbenchmarks-test/src/config/index.ts:49]: I think Jonas means we shouldn't have this validate method anymore. It was only in place to make sure `WS_ENABLED=true` while we didn't have rest.

**Review by boxhock — DISMISSED** [review]

**Comment by amit-momin** [comment]
I think the existing v3 adapters that haven't been upgraded past 0.19.0 are causing the legos test to fail with metrics conflicts. Think the culprit is specifically `bank-frick` for this one.

**Review by amit-momin — APPROVED** [review]

**Pull request merged**